### PR TITLE
Fi 46 quit command part3

### PR DESCRIPTION
**Pull request by jestebanpelaez18**
i tested all the possible cases, seems fine. I check with multiple clients, if the client that have created the channel quit, then the operator rights are give in the order they joined the channel. Then in the mod command if someone become operator of the channel we should add that client in the _operators vector and then also we need to add the part of the privmessage!

### CODE REVIEW

File: srcs/Channel.cpp [hunk 1]
```diff
@@ -54,7 +55,20 @@ void	Channel::addOperator(std::string operatorName)
 void	Channel::removeOperator(std::string operatorName)
 {
 	std::vector<std::string>::iterator it = std::find(_operators.begin(), _operators.end(), operatorName);
-	_operators.erase(it);
+	if(it != _operators.end())
+	{
+		_operators.erase(it);
+		if(!_operators.empty())
+			std::cout << _operators.back() << "is the new operator of the channeeeel" << std::endl;
```
Comment (ynslee) [inline srcs/Channel.cpp:62]: funny message but let's change channeeeel to channel

File: srcs/commands/Quit.cpp [hunk 2]
```diff
@@ -20,33 +20,50 @@
...
 
-// void sendMsgtoClients(std::string message, Client *Client, std::vector<std::string> &nick_names)
-// {
-//     for(size_t i = 0; i < nick_names.size(); i++)
-//     {
-//         if(Client != NULL && nick_names[i] == Client->getNickName())
-//             continue;
-//     }
-// }
+void sendMsgtoClients(std::string message, Client *client, Channel *channel)
```
Comment (ynslee) [inline srcs/commands/Quit.cpp:23]: let's add this to commands.hpp

File: srcs/Channel.cpp (hunk 1 above)
Comment (jestebanpelaez18) [reply srcs/Channel.cpp:62]: Done it!

File: srcs/commands/Quit.cpp (hunk 2 above)
Comment (jestebanpelaez18) [reply srcs/commands/Quit.cpp:23]: Done it!

**Review by ynslee — APPROVED** [review]

File: srcs/Channel.cpp [hunk 3]
```diff
@@ -54,7 +55,20 @@ void	Channel::addOperator(std::string operatorName)
...
+	if(it != _operators.end())
+	{
+		_operators.erase(it);
+		if(!_operators.empty())
+			std::cout << _operators.back() << "is the new operator of the channel" << std::endl;
+		else if(_clientOrder.size() > 1) 
+		{
+			_clientOrder.erase(_clientOrder.begin());
+			_clientOrder.front()->setIsOperator(true);
+			_clientOrder.front()->setMode("+o");
```
Comment (hussojo) [inline srcs/Channel.cpp]: let's not do this

**Review by hussojo — APPROVED** [review]

**Pull request merged**